lib/keymap-helpers.nix

```diff
@@ -77,16 +77,21 @@ in rec {
       (attrValues modes)
     );
 
-  # TODO: When `maps` will have to be deprecated (early December 2023), change this.
-  # mapOptionSubmodule = {...} (no need for options... except for 'optionalAction')
-  mapOptionSubmodule = with types;
+  mapOptionSubmodule = mkMapOptionSubmodule {};
+
+  mkMapOptionSubmodule = defaults: (with types;
     submodule {
       options = {
-        key = mkOption {
-          type = str;
-          description = "The key to map.";
-          example = "<C-m>";
-        };
+        key = mkOption ({
+            type = str;
+            description = "The key to map.";
+            example = "<C-m>";
+          }
+          // (
+            if defaults ? key
+            then {default = defaults.key;}
+            else {}
```

Comment:
lib.attrsets.optionalAttrs/lib.optionalAttrs ?

Reply:
Done (not a lot shorter though).